src/ORM/HasManyList.php

```diff
@@ -93,6 +93,12 @@ public function add($item)
         $item->$foreignKey = $foreignID;
 
         $item->write();
+
+        // Call addCallback, if applicable
+        if ($this->addCallback) {
```

Comment:
How about an `executeAddCallback` (name just a suggestion) method that does this for us so we don't have to repeat these lines of code.

```php
protected executeAddCallback($item, $extraFields = [])
{
    if ($this->addCallback) {
        return call_user_func($this->addCallback, $this, $item, $extraFields);
    }
}
```

Reply:
I think that's more noisy than simple inlined conditionals, particularly since we decided that `protected` methods are now part of our API surface - and we can't make it `private` since subclasses rely on it.